addons/mail/models/ir_attachment.py

```diff
@@ -10,6 +10,18 @@
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    @api.model
+    def check(self, mode, values=None):
+        super().check(mode, values=values)
+        if mode not in ('unlink', 'write') or not self or self.env.is_admin():
+            return
+        linked_messages = self.env['mail.message'].sudo().search([('attachment_ids', 'in', self.ids)])
```

Comment:
If you're the create_uid of these attachment records, can't we bypass this check?

Idea would be to save a request to mail.message, which can be a potential pain point, and say "ok you're the creator of these attachments, you can delete them".

I'm guessing it's the most common use case so could be nice to save some perf.

Reply:
Indeed, only "downside" is you can delete attachments used in other people's messages but in practice that should never happen as we copy attachments on send in every case that matters.